openedx/core/djangoapps/notifications/serializers.py

```diff
@@ -156,10 +156,69 @@ def update(self, instance, validated_data):
         return instance
 
 
+class UserNotificationChannelPreferenceUpdateSerializer(serializers.Serializer):
+    """
+    Serializer for user notification preferences update for an entire channel.
+    """
+
+    notification_app = serializers.CharField()
+    value = serializers.BooleanField()
+    notification_channel = serializers.CharField(required=False)
+
+    def validate(self, attrs):
+        """
+        Validation for notification preference update form
+        """
+        notification_app = attrs.get('notification_app')
+        notification_channel = attrs.get('notification_channel')
+
+        notification_app_config = self.instance.notification_preference_config
+
+        if not notification_channel:
+            raise ValidationError(
+                'notification_channel is required for notification_type.'
+            )
+
+        if not notification_app_config.get(notification_app, None):
+            raise ValidationError(
+                f'{notification_app} is not a valid notification app.'
+            )
+
+        if notification_channel and notification_channel not in get_notification_channels():
+            raise ValidationError(
+                f'{notification_channel} is not a valid notification channel.'
+            )
+
+        return attrs
+
+    def update(self, instance, validated_data):
+        """
+        Update notification preference config.
+        """
+        notification_app = validated_data.get('notification_app')
+        notification_channel = validated_data.get('notification_channel')
+        value = validated_data.get('value')
+        user_notification_preference_config = instance.notification_preference_config
+
+        for notification_type_name, notification_type_preferences in user_notification_preference_config[
+                notification_app]['notification_types'].items():
+            if notification_type_name not in user_notification_preference_config[notification_app]['non_editable']:
+                user_notification_preference_config[notification_app]['notification_types'][notification_type_name][
+                    notification_channel] = value
+            elif notification_channel not in user_notification_preference_config[notification_app]['non_editable'][
+                    notification_type_name]:
+                user_notification_preference_config[notification_app]['notification_types'][notification_type_name][
+                    notification_channel] = value
```

Comment:
nit: 
 - Using nested indexes reduces readability. 
 - You can update value in one condition

```
app_prefs = user_notification_preference_config[notification_app]
for notification_type_name, notification_type_preferences in app_prefs['notification_types'].items():
    non_editable_channels = app_prefs['non_editable'].get(notification_type_name, [])
    if channel not in non_editable_channels:
        app_prefs['notification_types'][notification_type_name][notification_channel] = value

```